pkg/collector/corechecks/snmp/config_metric.go

```diff
@@ -0,0 +1,210 @@
+			}
+			newFullIndex := strings.Join(newIndexes, ".")
+
+			tagValue, ok := columnValues[newFullIndex]
+			if !ok {
+				log.Debugf("index not found for column value: tag=%v, index=%v", metricTag.Tag, newFullIndex)
+				continue
+			}
+			strValue, err := tagValue.toString()
+			if err != nil {
+				log.Debugf("error converting tagValue (%#v) to string : %v", tagValue, err)
+				continue
+			}
+			rowTags = append(rowTags, metricTag.getTags(strValue)...)
+		}
+	}
+	return rowTags
+}
+
+func (m *metricsConfig) getSymbolTags() []string {
+	var symbolTags []string
+	for _, metricTag := range m.MetricTags {
+		symbolTags = append(symbolTags, metricTag.symbolTag)
+	}
+	return symbolTags
+}
+
+func (m *metricsConfig) isColumn() bool {
+	return len(m.Symbols) > 0
+}
+
+func (m *metricsConfig) isScalar() bool {
+	return m.Symbol.OID != "" && m.Symbol.Name != ""
+}
+
+func (mtc *metricTagConfig) getTags(value string) []string {
+	var tags []string
+	if mtc.Tag != "" {
+		tags = append(tags, mtc.Tag+":"+value)
+	} else if mtc.Match != "" {
+		if mtc.pattern == nil {
+			log.Warnf("match pattern must be present: match=%s", mtc.Match)
+			return tags
+		}
+		if mtc.pattern.MatchString(value) {
+			for key, val := range mtc.Tags {
+				normalizedTemplate := normalizeRegexReplaceValue(val)
+				replacedVal := regexReplaceValue(value, mtc.pattern, normalizedTemplate)
+				if replacedVal == "" {
+					log.Debugf("pattern `%v` failed to match `%v` with template `%v`", value, normalizedTemplate)
+					continue
+				}
+				tags = append(tags, key+":"+replacedVal)
+			}
+		}
+	}
+	return tags
+}
+
+func regexReplaceValue(value string, pattern *regexp.Regexp, normalizedTemplate string) string {
+	result := []byte{}
+	for _, submatches := range pattern.FindAllStringSubmatchIndex(value, 1) {
+		result = pattern.ExpandString(result, normalizedTemplate, value, submatches)
+	}
+	return string(result)
+}
+
+// normalizeRegexReplaceValue normalize regex value to keep compatibility with Python
+// Converts \1 into $1, \2 into $2, etc
+func normalizeRegexReplaceValue(val string) string {
+	re := regexp.MustCompile("\\\\(\\d+)")
+	return re.ReplaceAllString(val, "$$$1")
+}
+
+// transformIndex change a source index into a new index using a list of transform rules.
+// A transform rule has start/end fields, it is used to extract a subset of the source index.
+func transformIndex(indexes []string, transformRules []metricIndexTransform) []string {
+	var newIndex []string
+
+	for _, rule := range transformRules {
+		start := rule.Start
+		end := rule.End + 1
+		if end > uint(len(indexes)) {
+			return nil
+		}
+		newIndex = append(newIndex, indexes[start:end]...)
+	}
+	return newIndex
+}
+
+// normalizeMetrics converts legacy syntax to new syntax
+// 1/ converts old symbol syntax to new symbol syntax
+//    metric.Name and metric.OID info are moved to metric.Symbol.Name and metric.Symbol.OID
+func normalizeMetrics(metrics []metricsConfig) {
+	for i := range metrics {
```

Comment:
This will unfortunately lead to a copy each metric and we won't be modifying in place.

Learning this the hard way here: https://github.com/DataDog/datadog-agent/pull/6943/files#diff-e58a1523f3120ce9446a8908240ffca2d36a095cdf748d67721dd71495621f3bR231-R234